### PR TITLE
Add more exploratory analysis from discussion with collaborator

### PR DESCRIPTION
**Pull request by ajlee21**
This PR is adding and updating analyses to continue to explore the data to figure out what set of genes and what story to tell about core and accessory genes. 

Changes include:
1. Exploring accessory genes most related to exoU and exoS genes (`3_explore_exo_modules.ipynb`)
2. Update notebooks in `5_core_core_analysis/` directory to output annotations for all core genes, not sure those that are most and least stable
3. Add KEGG enrichment analysis to identify any pathways that are significantly enriched in our most and least stable gene sets (`4_KEGG_enrichment_of_stable_genes.ipynb`)
4. Update common DEG analysis after training two new VAE models using the PAO1 and PA14 RNA-seq compendia defined in this repository. This change is coupled with https://github.com/greenelab/generic-expression-patterns/pull/96. Note: I plan to make another PR to copy the necessary scripts from the generic-expression-patterns repo into this repo to use so that the analysis is all in one place. Ideally it would probably had made sense to have the scripts from the generic-expression-patterns repo be its own portable module, but this is the easiest clean way I can think to do this. (`1_common_DEGs_and_gene_groups.ipynb`)

### CODE REVIEW

**Comment by nrosed** [comment]
Look good to me! To be honest, I can't give in-depth feedback on your methodology/approach, but from what I understand it seems clear and well-motivated. Only very minor comments.

**Review by nrosed — APPROVED** [review]

File: 5_core_core_analysis/4_KEGG_enrichment_of_stable_genes.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+print(pao1_least_stable_enrichment.shape)
+pao1_least_stable_enrichment.sort_values(by="corrected p-value").head()
+
+# Save
+pao1_most_stable_enrichment.to_csv("pao1_most_stable_enrichment.tsv", sep="\t")
+pao1_least_stable_enrichment.to_csv("pao1_least_stable_enrichment.tsv", sep="\t")
+
+# **Takeaway:**
+# * There does not appear to be any enriched KEGG pathways in the least stable genes.
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply 5_core_core_analysis/4_KEGG_enrichment_of_stable_genes.py:151]: This is a good point. I'll add this. Thank you

**Comment by ajlee21** [comment]
Thank you for the comments. I know all the analyses are a bit hard to follow at the moment because there isn't a clear story just yet, its still in the works. So thanks for taking a look.
